core/src/main/scala/com/microsoft/azure/synapse/ml/core/contracts/Params.scala

```diff
@@ -205,3 +205,18 @@ trait HasGroupCol extends Params {
   /** @group getParam */
   def getGroupCol: String = $(groupCol)
 }
+
+trait HasExcludedFeatures extends Params {
+
+  /** The names of the feature columns to be excluded
+    *
+    * @group param
+    */
+  val excludedFeatures = new StringArrayParam(this, "excludedFeatureCols", "The names of the features to be excluded from training.")
+
+  /** @group setParam */
+  def setExcludedFeatures(value: Array[String]): this.type = set(excludedFeatures, value)
+
+  /** @group getParam */
+  def getExcludedFeatures: Array[String] = $(excludedFeatures)
+}
```

Comment:
Would it make sense to standardize this stuff against the HasFeatureCol API instead of this excluded features col API, this will hgelp you make this generic against any SparkML model

Reply:
Do you mean "HasFeaturesCol" API? it is referring to the "features" vector column. I don't think it's for our purpose here.

Or do you mean we add "HasFeatureCols" API to enable the function to "include which columns into features"? 
I think even if we have it, excludedFeatures API is still useful as sometimes we just want to include all columns except a few as features.

Reply:
Maybe we can chat the three of us to figure out what would be the best thing to do here

Reply:
will setup a quick sync, thanks!

Reply:
Added "HasFeatureCols" but haven't figured out a way to remove "HasNonFeatureCols", I need it to remove outcome column for treatment model, remove treatment column for outcome model.
@memoryz , any idea?